pkg/ipam/allocator/multipool/pool_allocator.go

```diff
@@ -94,18 +162,29 @@ func addrsInPrefix(p netip.Prefix) *big.Int {
 }
 
 type PoolAllocator struct {
-	logger *slog.Logger
-	mutex  lock.RWMutex
-	pools  map[string]cidrPool    // poolName -> pool
-	nodes  map[string]poolToCIDRs // nodeName -> pool -> cidrs
-	ready  bool
+	logger                    *slog.Logger
+	mutex                     lock.RWMutex
+	pools                     map[string]cidrPool    // poolName -> pool
+	nodes                     map[string]poolToCIDRs // nodeName -> pool -> cidrs
+	ready                     bool
+	metricsAPI                metricsAPI
+	CountAllocatedIPsInPrefix func(cidr *netip.Prefix) *big.Int
```

Comment:
Not sure why we need a function as a field of this struct. What is the reason behind this?

Reply:
But why a field in a struct and not a simple function in the `multipool` package?

Also, `CountAllocatedIPsInPrefix` is initialized in `NewPoolAllocator` to always returns 0 and never overwritten, so it will always return 0. What's the point?